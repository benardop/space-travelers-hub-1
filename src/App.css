@import url('https://fonts.googleapis.com/css2?family=Poppins&display=swap');

.App {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: 'Poppins', sans-serif;
}

:root {
  --color-blue: #157af6;
}
