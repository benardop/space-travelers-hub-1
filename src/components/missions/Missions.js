import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchMissionsApi } from '../../redux/missions/MissionsSlice';
import Missioncard from './Missioncard';
import '../stylesheets/Missioncard.css';

const Missionlist = () => {
  const dispatch = useDispatch();
  const mission = useSelector((state) => state.missions);
  useEffect(() => {
    dispatch(fetchMissionsApi());
  }, []);
  return (
    <div>
      <div className="heading">
        <div className="left">Mission</div>
        <div className="center">Description</div>
        <div className="right">Status</div>
      </div>
      { mission.map((item) => (
        <Missioncard
          key={item.missionid}
          missionname={item.missionName}
          description={item.missionDesc}
          id={item.missionid}
          missionjoin={item.missionjoin}
        />
      )) }
    </div>
  );
};

export default Missionlist;
