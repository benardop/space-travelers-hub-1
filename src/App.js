import { Route, Routes } from 'react-router-dom';
import './App.css';
import Header from './components/Header';
import Missions from './components/missions/Missions';
import Myprofile from './components/myprofile/Myprofile';
import Rockets from './components/rockets/Rockets';

function App() {
  return (
    <div className="App">
      <Header />
      <Routes>
        <Route path="/" element={<Rockets />} />
        <Route path="missions" element={<Missions />} />
        <Route path="myprofile" element={<Myprofile />} />
      </Routes>
    </div>
  );
}

export default App;
