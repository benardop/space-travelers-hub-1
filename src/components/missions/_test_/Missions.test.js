import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from '../../../App';
import store from '../../../redux/configureStore';
import Missions from '../Missions';
import Missioncard from '../Missioncard';

// missions snapshots
const MockMissions = () => (
  <Provider store={store}>
    <Missions />
  </Provider>
);

it('matches snapshot', () => {
  const tree = renderer.create(
    <MockMissions />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

// test missions link from the app section
const MockApp = () => (
  <Provider store={store}>
    <BrowserRouter>
      <App />
    </BrowserRouter>
  </Provider>
);

describe('when you click the missions link', () => {
  it('should render missions page content', () => {
    render(
      (<MockApp />),
    );
    userEvent.click(screen.getByRole('link', { name: /missions/i }));
    expect(screen.getByText(/description/i)).toBeInTheDocument();
  });
});

// check button change text
describe('when you click button it should change to leave mission', () => {
  it('should render missions page content', () => {
    render(
      (<MockMissioncard />),
    );
    userEvent.click(screen.getByRole('button', { name: /leave mission/i }));
    expect(screen.getByText(/Active Member/)).toBeInTheDocument();
  });
});

// test the mission card
const missioncard = [
  {
    missionid: '9D1B7E0',
    missionName: 'Thaicom',
    missionDesc:
    'Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.',
    missionweblink: 'https://en.wikipedia.org/wiki/Thaicom',
    missionjoin: true,
  }];

const MockMissioncard = () => (
  <Provider store={store}>
    <Missioncard missioncard={missioncard} />
  </Provider>
);

test('renders data in the card as from API', () => {
  const tree = renderer.create(<MockMissioncard />).toJSON();
  expect(tree).toMatchSnapshot();
});
