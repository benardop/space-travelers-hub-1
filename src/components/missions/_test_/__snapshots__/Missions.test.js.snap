// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    className="heading"
  >
    <div
      className="left"
    >
      Mission
    </div>
    <div
      className="center"
    >
      Description
    </div>
    <div
      className="right"
    >
      Status
    </div>
  </div>
</div>
`;

exports[`renders data in the card as from API 1`] = `
<div
  className="mission-card"
>
  <div
    className="mission"
  >
    <div
      className="tbody"
    >
      <div
        className="left"
      />
      <div
        className="center"
      />
      <div
        className="right"
      >
        <span
          className="badge"
        >
          Active Member
        </span>
        <button
          className="join-btn"
          onClick={[Function]}
          type="submit"
        >
          Leave Mission
        </button>
      </div>
    </div>
  </div>
</div>
`;
